errors.js

```diff
@@ -0,0 +1,72 @@
+class AppError extends Error {
+  constructor(message) {
+    super(message);
+
+    this.name = "";
+    this.code = 0;
+  }
+}
+
+class BadRequestError extends AppError {
+  constructor(message) {
```

Comment:
I quite like this idea, I would even consider having default values for the messages that can optionally be overwritten